systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java

```diff
@@ -1081,212 +1075,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
-            .withProducerName(testStorage.getProducerName())
-            .withConsumerName(testStorage.getConsumerName())
-            .withMessageCount(testStorage.getMessageCount())
-            .build();
+        // key-value pairs modification and addition of user specified labels for kafka CR metadata
+        final String firstKafkaLabelValueModified = "first-kafka-label-value-modified";
+        final String thirdKafkaLabelKey = "third-kafka-label-key";
+        final String thirdKafkaLabelValue = "third-kafka-label-value";
+        customSpecifiedLabels.replace(firstKafkaLabelKey, firstKafkaLabelValueModified);
+        customSpecifiedLabels.put(thirdKafkaLabelKey, thirdKafkaLabelValue);
+        LOGGER.info("New values of labels which are to modify Kafka CR after their replacement and addition of new one are following {}", customSpecifiedLabels);
 
-        LOGGER.info("---> PODS <---");
+        // key-value pair modification of user specified label in managed PVCs
+        final String pvcLabelOrAnnotationValueModified = "pvc-label-value-modified";
+        customSpecifiedLabelOrAnnotationPvc.replace(pvcLabelOrAnnotationKey, pvcLabelOrAnnotationValueModified);
+        LOGGER.info("New values of labels which are to modify label and annotation of PVC present in Kafka CR, with following values {}", customSpecifiedLabelOrAnnotationPvc);
 
-        List<Pod> pods = kubeClient(testStorage.getNamespaceName()).listPods(testStorage.getNamespaceName(), testStorage.getClusterName()).stream()
-                .filter(pod -> pod.getMetadata().getName().startsWith(testStorage.getClusterName()))
-                .collect(Collectors.toList());
+        LOGGER.info("Edit Kafka labels in Kafka CR,as well as labels, and annotations of PVCs");
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), resource -> {
+            for (Map.Entry<String, String> label : customSpecifiedLabels.entrySet()) {
+                resource.getMetadata().getLabels().put(label.getKey(), label.getValue());
+            }
+            resource.getSpec().getKafka().getTemplate().getPersistentVolumeClaim().getMetadata().setLabels(customSpecifiedLabelOrAnnotationPvc);
+            resource.getSpec().getKafka().getTemplate().getPersistentVolumeClaim().getMetadata().setAnnotations(customSpecifiedLabelOrAnnotationPvc);
+            resource.getSpec().getZookeeper().getTemplate().getPersistentVolumeClaim().getMetadata().setLabels(customSpecifiedLabelOrAnnotationPvc);
+            resource.getSpec().getZookeeper().getTemplate().getPersistentVolumeClaim().getMetadata().setAnnotations(customSpecifiedLabelOrAnnotationPvc);
+        }, testStorage.getNamespaceName());
 
-        for (Pod pod : pods) {
-            LOGGER.info("Getting labels from {} pod", pod.getMetadata().getName());
-            verifyAppLabels(pod.getMetadata().getLabels());
-        }
+        LOGGER.info("Wait for rollout of Zookeeper and Kafka");
```

Comment:
for rolling update instead of rollout?

Reply:
👍 changed. both works in my case but your suggestion is more specific which is good.